modules/subschema/src/main/scala/com.snowplowanalytics/iglu.schemaddl/jsonschema/subschema/package.scala

```diff
@@ -12,23 +14,34 @@ package object subschema {
 
   def canonicalize(s: Schema): Schema =
     s match {
-      case s if s.`type` == Some(Integer) => s.copy(`type` = Some(Number))
-      case s if s.`type` == Some(Number)  => s
-      case s if s.`type` == Some(String)  => s
+      case s if s.`type`.contains(Null)    => s
+      case s if s.`type`.contains(Boolean) =>
+        if (s.`enum`.isDefined) s else s.copy(`enum` = Some(Enum(List(Json.True, Json.False))))
+      case s if s.`type`.contains(Integer) => s.copy(`type` = Some(Number))
+      case s if s.`type`.contains(Number)  => s
+      case s if s.`type`.contains(String)  => s
       case _ => s
     }
 
 
   def simplify(s: Schema): Schema = s
 
   def isSubType(s1: Schema, s2: Schema): Compatibility = (s1, s2) match {
-    case (s1, s2) if s1.`type` == Some(Number) && s1.`type` == s2.`type` => isNumberSubType(s1, s2)
-    case (s1, s2) if s1.`type` == Some(String) && s1.`type` == s2.`type` => isStringSubType(s1, s2)
+    case (s1, s2) if s1.`type`.contains(Null) && s1.`type` == s2.`type`    => Compatible
+    case (s1, s2) if s1.`type`.contains(Boolean) && s1.`type` == s2.`type` => isBooleanSubType(s1, s2)
+    case (s1, s2) if s1.`type`.contains(Number) && s1.`type` == s2.`type`  => isNumberSubType(s1, s2)
+    case (s1, s2) if s1.`type`.contains(String) && s1.`type` == s2.`type`  => isStringSubType(s1, s2)
     case _ => Undecidable
   }
 
   def stripAnchors(r: String): String = r.stripPrefix("^").stripSuffix("$")
 
+  def isBooleanSubType(s1: Schema, s2: Schema): Compatibility =
+    (s1.`enum`.map(_.value.toSet), s2.`enum`.map(_.value.toSet)) match {
+      case (maybeXe1, Some(xe2)) if !maybeXe1.getOrElse(Set(Json.True, Json.False)).subsetOf(xe2) => Incompatible
```

Comment:
The only possible way that makes a boolean schema NOT a subset of another one is when the (explicit or inferred) set of possible values is not a subset of all possible values of the other schema. All other cases make the first schema a subset of the second schema

Reply:
Makes sense!